### PR TITLE
OpenTelemetry Protocol with Apache Arrow Receiver component

### PR DESCRIPTION
**Pull request by jmacd**
Description:

This is the same code as OTel-Arrow at [14c63d1eaac7c53585e6b9195d09f1f9703869ed](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32015/commits/c73fd9243d6b375b27782ee5d0a1e84b5d89e509), at which point lint fixes required for this repository were applied. Only import statements change here, to match the host repository.

Link to tracking Issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26491

Testing: Test coverage is approximately 90%.

Documentation: I double-checked and the existing README, no changes required.

### CODE REVIEW

File: receiver/otelarrowreceiver/factory.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ const (
 	defaultMemoryLimitMiB = 128
 )
 
-// NewFactory creates a new OTel-Arrow receiver factory.
+// NewFactory creates a new OTLP receiver factory.
```
Comment (codeboten) [inline receiver/otelarrowreceiver/factory.go:25]: this seems incorrect

File: receiver/otelarrowreceiver/config_test.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func TestUnmarshalConfig(t *testing.T) {
 				GRPC: configgrpc.ServerConfig{
 					NetAddr: confignet.AddrConfig{
 						Endpoint:  "0.0.0.0:4317",
-						Transport: confignet.TransportTypeTCP,
+						Transport: "tcp",
```
Comment (codeboten) [inline receiver/otelarrowreceiver/config_test.go:54]: Not a big deal, but is this change needed?

File: receiver/otelarrowreceiver/config_test.go [hunk 3]
```diff
@@ -100,17 +100,25 @@ func TestUnmarshalConfigUnix(t *testing.T) {
 				GRPC: configgrpc.ServerConfig{
 					NetAddr: confignet.AddrConfig{
 						Endpoint:  "/tmp/grpc_otlp.sock",
-						Transport: confignet.TransportTypeUnix,
+						Transport: "unix",
```
Comment (codeboten) [inline receiver/otelarrowreceiver/config_test.go:103]: same question as above here

**Review by codeboten — COMMENTED** [review]
thanks for continuing to push this work forward @jmacd, just a couple of questions. Please resolve the conflicts and we can move this forward.

**Comment by jmacd** [comment]
Will re-open this.

File: cmd/otelcontribcol/go.mod [hunk 4]
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.21.0
 
-toolchain go1.21.9
+toolchain go1.22.2
```
Comment (codeboten) [inline cmd/otelcontribcol/go.mod:7]:
```suggestion
toolchain go1.21.9
```

**Review by codeboten — APPROVED** [review]

**Comment by codeboten** [comment]
Needs a `make genotelcontribcol`

**Review by moh-osman3 — APPROVED** [review]

**Comment by jmacd** [comment]
> Generated code is out of date, please run "make genotelcontribcol" and commit the changes in this PR.

Somehow `make genotelcontribcol` is doing nothing for me, locally. 😭

**Comment by codeboten** [comment]
@jmacd hmmm.... the same thing on my end, will take a look to see what's going on

**Comment by codeboten** [comment]
@jmacd the problem is caused by CICD using go 1.21.9 when running make genotelcontribcol, which causes a diff in the `toolchain`:

```
--- a/cmd/otelcontribcol/go.mod
+++ b/cmd/otelcontribcol/go.mod
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.21.0
 
-toolchain go1.22.2
+toolchain go1.21.9
```

I've pushed the change, but if you run make genotelcontribcol locally, it will be set back to 1.22.2 😮‍💨

**Comment by codeboten** [comment]
Long story short, if you can resolve the last conflict, we can get this merged and never think about it again. Sadly, this new `toolchain` directive is causing all sorts of problems :(
